packages/db/tests/models/test-granule-model.js

```diff
@@ -49,21 +57,251 @@ test.after.always(async (t) => {
+    knex,
+    granule
+  );
+
+  await granuleExecutionHistoryPgModel.create(knex, {
+    granule_cumulus_id: granuleCumulusId,
+    execution_cumulus_id: executionCumulusId,
+  });
+
+  const [newExecutionCumulusId] = await executionPgModel.create(
+    knex,
+    fakeExecutionRecordFactory()
+  );
+
+  await granuleExecutionHistoryPgModel.create(knex, {
+    granule_cumulus_id: granuleCumulusId,
+    execution_cumulus_id: newExecutionCumulusId,
+  });
+
+  const [granuleWithHistory] = await granulePgModel.getWithExecutionHistory(
+    knex,
+    granule
+  );
+
+  t.like(
+    granuleWithHistory,
+    {
+      ...granule,
+      cumulus_id: granuleCumulusId,
+      execution_cumulus_ids: [executionCumulusId, newExecutionCumulusId],
+    }
+  );
+});
+
+test('GranulePgModel.createWithExecutionHistory() creates a new granule with execution history', async (t) => {
+  const {
+    knex,
+    granulePgModel,
+    collectionCumulusId,
+    executionCumulusId,
+    granuleExecutionHistoryPgModel,
+  } = t.context;
+
+  const granule = fakeGranuleRecordFactory({
+    collection_cumulus_id: collectionCumulusId,
+    status: 'running',
+  });
+
+  const [granuleCumulusId] = await granulePgModel.createWithExecutionHistory(
+    knex,
+    granule,
+    executionCumulusId
+  );
+
+  const granuleRecord = await granulePgModel.get(
+    knex,
+    granule
+  );
+
+  t.like(
+    granuleRecord,
+    {
+      ...granule,
+      cumulus_id: granuleCumulusId,
+    }
+  );
+  t.deepEqual(
+    await granuleExecutionHistoryPgModel.search(
+      knex,
+      { granule_cumulus_id: granuleCumulusId }
+    ),
+    [{
+      granule_cumulus_id: Number.parseInt(granuleCumulusId, 10),
+      execution_cumulus_id: executionCumulusId,
+    }]
+  );
+});
+
+test('GranulePgModel.createWithExecutionHistory() works with a transaction', async (t) => {
+  const {
+    knex,
+    granulePgModel,
+    collectionCumulusId,
+    executionCumulusId,
+    granuleExecutionHistoryPgModel,
+  } = t.context;
+
+  const granule = fakeGranuleRecordFactory({
+    collection_cumulus_id: collectionCumulusId,
+    status: 'running',
+  });
+
+  const [granuleCumulusId] = await knex.transaction(
+    (trx) =>
+      granulePgModel.createWithExecutionHistory(
+        trx,
+        granule,
+        executionCumulusId
+      )
+  );
+
+  const granuleRecord = await granulePgModel.get(
+    knex,
+    granule
+  );
+
+  t.like(
+    granuleRecord,
+    {
+      ...granule,
+      cumulus_id: granuleCumulusId,
+    }
+  );
+  t.deepEqual(
+    await granuleExecutionHistoryPgModel.search(
+      knex,
+      { granule_cumulus_id: granuleCumulusId }
+    ),
+    [{
+      granule_cumulus_id: Number.parseInt(granuleCumulusId, 10),
+      execution_cumulus_id: executionCumulusId,
+    }]
+  );
+});
+
+test('GranulePgModel.createWithExecutionHistory() does not commit granule or execution history if execution history write fails', async (t) => {
```

Comment:
See comments above - I believe this behavior depends on if a transaction is passed in or not.     

This probably shouldn't be the desired behavior, but if it is, this test title should note the distinction (e.g.  "GranulePgModel.createWithExecutionHistory() when passed a knex transaction does not..."

Reply:
Oh, yeah, it is dependent on a transaction. Because unless you're using a transaction then you can't have rollback behavior

Reply:
https://github.com/nasa/cumulus/pull/2091/commits/baedd0f61b299ff87eb434d160f8e3f56f4a10ab